### PR TITLE
fix: fixed an issue that would cause a malformed response if no update types exist yet

### PR DESCRIPTION
**Pull request by mzedel**
Ticket: None
Changelog: None

### CODE REVIEW

File: store/mongo/datastore_mongo_releases.go [hunk 1]
```diff
@@ -279,6 +280,9 @@ func (db *DataStoreMongo) GetUpdateTypes(ctx context.Context) ([]string, error)
 	}
 	var updateTypes updateType
 	err := result.Decode(&updateTypes)
+	if err == mongo.ErrNoDocuments {
+		return []string{}, nil
+	}
```
Comment (merlin-northern) [inline store/mongo/datastore_mongo_releases.go:285]: so you want no error on no documents. ok.

File: tests/tests/test_release_v2.py [hunk 2]
```diff
@@ -142,3 +142,24 @@ def test_get_all_releases_types(self, mongo, cli):
...
+                + f"/api/management/v2/deployments/releases/all/types"
+            )
+
+            r = requests.get(
+                get_types_url,
+                verify=False,
+                headers={"Authorization": "Bearer foo"},
+            )
+            types = json.loads(r.text)
+            assert types == []
```
Comment (merlin-northern) [inline tests/tests/test_release_v2.py:165]: we could probably assert here that there was no error (HTTP code success)

**Review by merlin-northern — APPROVED** [review]
misc comment in the tests only. thank you!

File: store/mongo/datastore_mongo_releases.go (hunk 1 above)
Comment (mzedel) [reply store/mongo/datastore_mongo_releases.go:285]: not quite... the API spec promises an array, as the endpoint is supposed to respond with an array and an empty array is a valid response if no items to populate the list exist...

What I _want_ is for the backend to validate the backend endpoints with the API specs.

File: tests/tests/test_release_v2.py (hunk 2 above)
Comment (mzedel) [reply tests/tests/test_release_v2.py:165]: sure, but I wanted to align with the test right above (where I took this from).

**Comment by mzedel** [comment]
@mender-test-bot start pipeline --fast

File: store/mongo/datastore_mongo_releases.go (hunk 1 above)
Comment (mzedel) [reply store/mongo/datastore_mongo_releases.go:285]: no: https://github.com/mendersoftware/deployments/blob/master/api/http/api_deployments_releases.go#L192-L196

Comment (merlin-northern) [reply store/mongo/datastore_mongo_releases.go:285]: ok then!

**Pull request merged**